lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -448,6 +449,242 @@ class ConvertAtenLinearOp : public OpConversionPattern<AtenLinearOp> {
+    }
+
+    auto getConvOutputDim = [&](Value lhDim, Value rhDim, Value padding,
+                                Value stride, Value dilation,
+                                bool valid) -> Value {
+      if (valid) {
+        return torch_to_linalg::getOutputDimForConvOps(
+            rewriter, loc, lhDim, ci0, dilation, castIndexToInt(rhDim), stride);
+      }
+      // TODO: dilation, stride
+      Value c1 =
+          rewriter.create<arith::ConstantOp>(loc, rewriter.getIndexAttr(1));
+      Value res = rewriter.create<arith::AddIOp>(loc, lhDim, rhDim);
+      return rewriter.create<arith::SubIOp>(loc, res, c1);
+    };
+    auto convolve = [&](Value lhs, Value rhs, SmallVector<Value> &lhDims,
+                        SmallVector<Value> &rhDims, bool valid) -> Value {
+      Value paddedLhs = (valid ? lhs
+                               : torch_to_linalg::getPaddedTensor(
+                                     op, rewriter, lhs, paddingIncludingNC,
+                                     paddingIncludingNC, c0));
+
+      SmallVector<Value> outDims{lhDims[0], rhDims[0]};
+      for (size_t i = 0; i < inRank - 2; i++)
+        outDims.push_back(
+            getConvOutputDim(lhDims[i + 2], rhDims[i + 2], paddingIntValues[i],
+                             strideIntValues[i], dilationIntValues[i], valid));
+
+      Value initTensor =
+          rewriter.create<linalg::InitTensorOp>(loc, outDims, elementType);
+
+      Value biasInitTensor;
+      biasInitTensor =
+          rewriter.create<linalg::FillOp>(loc, c0, initTensor).getResult(0);
+
+      // TODO: add 1D and 3D case
+      Value conv =
+          rewriter
+              .create<linalg::Conv2DNchwFchwOp>(
+                  loc, biasInitTensor.getType(), ValueRange{paddedLhs, rhs},
+                  biasInitTensor, stridesAttr, dilationAttr)
+              .getResult(0);
+
+      return conv;
+    };
+
+    // Convolution between grad_output and rotated weight (input gradient)
+    auto rotate = [&](Value &input, uint64_t startDim,
+                      uint64_t endDim) -> Value {
+      RankedTensorType tType = input.getType().cast<RankedTensorType>();
+      SmallVector<Value> tSizes = getTensorSizes(rewriter, loc, input);
+
+      Value c1 =
+          rewriter.create<arith::ConstantOp>(loc, rewriter.getIndexAttr(1));
+      SmallVector<Value> flip;
+      for (size_t i = 0; i < tSizes.size(); i++)
+        flip.push_back(rewriter.create<arith::SubIOp>(loc, tSizes[i], c1));
+
+      SmallVector<StringRef> iteratorTypes(tSizes.size(), "parallel");
+      SmallVector<AffineExpr> exprs;
+      for (size_t i = 0; i < tSizes.size(); i++)
+        exprs.push_back(rewriter.getAffineDimExpr(i));
+      SmallVector<AffineMap> indexingMaps =
+          AffineMap::inferFromExprList({exprs, exprs});
+      Value outTensor = rewriter.create<linalg::InitTensorOp>(
+          loc, tSizes, tType.getElementType());
+
+      return rewriter
+          .create<linalg::GenericOp>(
```

Comment:
nit: if you move this function

https://github.com/llvm/torch-mlir/blob/9ec0683e928bdd1acea9aeb025ed542332fab902/lib/Conversion/TorchToLinalg/Uncategorized.cpp#L38

to one of the utils files, you can use it to avoid writing all the `iteratorTypes`, `indexingMaps`, ... boilerplate code